flutter_vlc_player/lib/src/vlc_player_controller.dart

```diff
@@ -141,20 +141,20 @@ class VlcPlayerController extends ValueNotifier<VlcPlayerValue> {
     }
 
     _lifeCycleObserver = VlcAppLifeCycleObserver(this);
-    _lifeCycleObserver.initialize();
+    _lifeCycleObserver!.initialize();
     _creatingCompleter = Completer<void>();
 
     await vlcPlayerPlatform.create(
-      viewId: _viewId,
+      viewId: _viewId!,
       uri: dataSource,
       type: dataSourceType,
       package: package,
-      hwAcc: hwAcc ?? HwAcc.AUTO,
-      autoPlay: autoPlay ?? true,
+      hwAcc: hwAcc,
+      autoPlay: autoPlay,
       options: options,
     );
 
-    _creatingCompleter.complete(null);
+    _creatingCompleter!.complete(null);
```

Comment:
We could probably avoid using a Completer here.
Given `Future<void>? _viewCreationFuture;` is a field in the enclosing class.
```dart
    _viewCreationFuture = vlcPlayerPlatform.create(
      viewId: _viewId,
      uri: dataSource,
      type: dataSourceType,
      package: package,
      hwAcc: hwAcc,
      autoPlay: autoPlay,
      options: options,
    );
    await _viewCreationFuture;
```
The future can be awaited elsewhere, if necessary.

Reply:
Resolved